partiql-logical-planner/src/lib.rs

```diff
@@ -0,0 +1,123 @@
+use crate::lower::AstToLogical;
+use crate::name_resolver::NameResolver;
+use partiql_ast::ast;
+use partiql_logical as logical;
+use partiql_parser::Parsed;
+
+mod lower;
+mod name_resolver;
+
+// TODO better encapsulate and add error types.
+pub fn lower(parsed: &Parsed) -> logical::LogicalPlan<logical::BindingsOp> {
+    if let ast::Expr::Query(q) = parsed.ast.as_ref() {
+        let mut resolver = NameResolver::default();
+        let resolver = resolver.resolve(q);
+
+        let planner = AstToLogical::new(resolver);
+        planner.lower_query(q)
+    } else {
+        panic!("wrong expr type");
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use assert_matches::assert_matches;
+
+    use partiql_eval::env::basic::MapBindings;
+
+    use partiql_eval::plan;
+
+    use partiql_logical as logical;
+    use partiql_logical::{BindingsOp, LogicalPlan};
+    use partiql_parser::{Parsed, Parser};
+    use partiql_value::{partiql_bag, partiql_tuple, Value};
+    use partiql_value::{Bag, Tuple};
+
+    #[track_caller]
+    fn parse(text: &str) -> Parsed {
+        Parser::default().parse(text).unwrap()
+    }
+
+    #[track_caller]
+    fn lower(parsed: &Parsed) -> logical::LogicalPlan<logical::BindingsOp> {
+        super::lower(parsed)
+    }
+
+    #[track_caller]
+    fn evaluate(logical: LogicalPlan<BindingsOp>, bindings: MapBindings<Value>) -> Value {
+        let planner = plan::EvaluatorPlanner;
+
+        let mut plan = planner.compile(&logical);
+        println!("{}", plan.to_dot_graph());
+
+        if let Ok(out) = plan.execute_mut(bindings) {
+            out.result
+        } else {
+            Value::Missing
+        }
+    }
+
+    #[track_caller]
+    fn evalute_query(query: &str) -> Value {
```

Comment:
nit: typo `evalute_query` -> `evaluate_query`. Also in the call sites on L109 and L116